### ISSUE TITLE
Where will the DID contexts(s) live?

### ISSUE THREAD

**Issue by dmitrizagidulin** [w3c/did-core#5]
Will the eventual `https://www.w3.org/2019/did/v1` and the current `https://w3id.org/did/v0.11` context for DIDs be version controlled in this repository now, or will it continue to reside at the CCG repo?

If the former, I would propose to pull over [PR #272 - Add a contexts/README.md to explain the DID context versioning scheme](https://github.com/w3c-ccg/did-spec/pull/272) over to this repo as well, since the topic of `@context` versioning may be confusing to the wider audience.

**Comment by OR13** [comment]
I recommend that both the human documentation and the JSON-LD contexts be hosted under this repo, and that we minimize use of / or links to 3rd party services which may have the ability to alter documentation or contexts in ways that impact usability or security.

It will also be easer to ensure documentation is up to date if a single PR can fix both the context and the human readable content.

Understanding of context processing is a requirement for understanding the security and usability of DIDs, I'm very much in favor of centralizing that trust here, I think many people don't realize how awesome JSON-LD is, and since DID Documents are JSON-LD, we have a responsibility to be up front about the security implications of working with JSON-LD.

**Comment by iherman** [comment]
My question/comment is on the URL used for the context file. At the moment, it is`https://www.w3.org/2019/did/v1`. I do not know whether, by now, this is cast in concrete; in many cases such context files (and possible namespace documents if separate terms are to be properly defined in RDF, too) are put under `/ns` at W3C. Like, for example, `http://www.w3.org/ns/anno.jsonld` for annotations or `https://www.w3.org/ns/pub-context` for publication manifests. (I know that the VC context file is under `/2018/`, though).

If this is not yet too late, I would propose to move the context file to `https://www.w3.org/ns/did/v1`.

Whichever way we go, setting up a redirect from that URL to a github repo file shouldn't be a problem. This is how, for example, `https://www.w3.org/ns/pub-context` is managed (although in that case, the context file is managed in a separate repo on github).

**Comment by dmitrizagidulin** [comment]
As I've tried to argue in the [DID context versioning PR](https://github.com/w3c-ccg/did-spec/pull/272/files), versioning the context is highly desirable (and even necessary, in our situation).

**Comment by OR13** [comment]
without versioning, how can we offline these contexts? won't that lead to lots of confusion over which offline context was used? 

I'm in favor of datespace over namespace, appears to add more useful information than `ns`.

**Comment by iherman** [comment]
This issue was discussed in [a meeting](https://www.w3.org/2019/did-wg/Meetings/Minutes/2019-10-15-did#section4).

- No actions or resolutions
<details><summary><i class="differentiate">View the transcript</i></summary>
5.1. context hosting #5<br/>
Manu Sporny: See Issue #5<br/>
Brent Zundel: Opinions?<br/>
Ivan Herman: There are two issues. Where and what URL.<br/>
… The usual place is w3.org/ns/…<br/>
… I propose we use the same method. The document can be stored elsewhere and have a redirect.<br/>
Manu Sporny: <a href="https://w3c.github.io/vc-data-model/#example-1-a-simple-example-of-a-verifiable-credential">https://w3c.github.io/vc-data-model/#example-1-a-simple-example-of-a-verifiable-credential</a><br/>
Manu Sporny: We had this discussion in VCWG.<br/>
… We decided to not use ns because of the emotional response.<br/>
Manu Sporny: <a href="https://www.w3.org/2018/credentials/v1">https://www.w3.org/2018/credentials/v1</a><br/>
Manu Sporny: There is not a redirect.<br/>
… During working drafts we redirected to the github updated version.<br/>
… The other thing was say in the spec that we will not change the context after the spec is finalized.<br/>
… This allows caching or hardcoding by developers.<br/>
… I hope we can use the same approach.<br/>
Ivan Herman: Whether we use ns or not: I find the date space more difficult personally<br/>
… but I won’t lie down the road over this.<br/>
… Redirect to github during development is my intention also.<br/>
… After we go to recommendation, it should be fixed.<br/>
Ivan Herman: I don’t know if we need to discuss versioning right now.<br/>
Phil Archer: The discusion re: persistant URL is of interest to me.<br/>
Phil Archer: See W3C Peristence policy<br/>
Phil Archer: The date space may not persist.<br/>
… Version numbers are problematic, ask DanBri and the Foaf URL<br/>
Amy Guy: I wanted to remind people that having the context on github made actual use difficult.<br/>
Orie Steele: +1 for hosting on github pages<br/>
Dmitri Zagidulin: Can we please host at github with a proper file extension.<br/>
Manu Sporny: Versioning must be accounted for.<br/>
… Most problems were with the hosting not the actual developers.<br/>
… Extensions and Mime types in particular.<br/>
Dmitri Zagidulin: (would like to remind everyone about the writeup / explainer about versioning contexts, over at <a href="https://github.com/w3c-ccg/did-spec/pull/272">https://github.com/w3c-ccg/did-spec/pull/272</a> )<br/>
Manu Sporny: During development will create a moving target necessarily.<br/>
… VCWG had the same debates.<br/>
… Can we summarize where we have consensus?<br/>
Brent Zundel: Can we wrap up?<br/>
Justin Richer: Some people will see the URLs as just a place for comparing.<br/>
Manu Sporny: +1 to Justin_R — yes, this isn’t a “purely Linked Data” world we’re working in here…<br/>
Manu Sporny: also – just to be clear… I’d personally be opposed to /ns/ URL and non-versioned (and we have good reasons for that)… I’ll have to see what our corporate position is.<br/>
Ivan Herman: Amy confirmed that redirection can be set up with proper types, so that is not a problem<br/>
Brent Zundel: Let’s move discussion to the issue.<br/>
Manu Sporny: I think we have consensus around hosting at W3C. Versioning on the github.<br/>
… No consensus on ns or versioning.<br/>
… I will summarize in the issue.<br/>
Manu Sporny: I summarized context issue here – <a href="https://github.com/w3c/did-spec/issues/5#issuecomment-542269659">https://github.com/w3c/did-spec/issues/5#issuecomment-542269659</a><br/>
</details>

**Comment by msporny** [comment]
> We didn't have consensus on:
> * What the URL will be, whether or it will be hosted at w3.org/ns/ or a date-space

For whatever reason, there is a non-trivial number of technologists that go apoplectic when we even mention the word "namespace" and divert the discussion into all the reasons "namespaces were a horrible idea for XML, a horrible idea for RDF, and a horrible idea now." We should avoid putting people using this technology into the position of having to have that discussion with anyone, because there are no winners when you trigger that discussion. So, let's not trigger that discussion by using the /ns/ namespace. :)

There are also other reasons having to do with versioning that we may want to avoid the /ns/ base URL elaborated upon below.

> We didn't have consensus on:
> * Whether or not we'll version the context.

We spent A LOT of time on this exact question in the Verifiable Claims WG. I went off looking for the links/threads and found 9 out of many more of them... and it's all hard to follow. So, I'm going to attempt to summarize here.

The Verifiable Credentials specification [states the following](https://www.w3.org/TR/vc-data-model/#contexts) about the `@context`:

> The value of the `@context` property MUST be an ordered set where the first item is a URI with the value https://www.w3.org/2018/credentials/v1. For reference, a copy of the base context is provided in Appendix § B. Base Context. Subsequent items in the array MUST express context information and be composed of any combination of URIs or objects. It is RECOMMENDED that each URI in the @context be one which, if dereferenced, results in a document containing machine-readable information about the `@context`.

> NOTE: Though this specification requires that a `@context` property be present, it is not required that the value of the `@context` property be processed using JSON-LD. This is to support processing using plain JSON libraries, such as those that might be used when the verifiable credential is encoded as a JWT. All libraries or processors MUST ensure that the order of the values in the `@context` property is what is expected for the specific application. Libraries or processors that support JSON-LD can process the `@context` property using full JSON-LD processing as expected.

> The data available at https://www.w3.org/2018/credentials/v1 is a static document that is never updated and SHOULD be downloaded and cached. The associated human-readable vocabulary document for the Verifiable Credentials Data Model is available at https://www.w3.org/2018/credentials. This concept is further expanded on in Section § 5.3 Extensibility.

That text was hard won and represents a best practice when attempting to build bridges between the Linked Data community and those that don't want to use (but still stay compatible with) Linked Data (which is an important subset of the community working on VCs and DIDs).

Fundamentally, we need to version the context *and* we need to year-stamp the URL... here's why:

We need to version the context `.../did/v1` because people will want a frozen in time document that they can refer to, cache, and hard code into their applications. People that don't want to do JSON-LD processing will use that string to trigger certain processing modes in their application.

We need to year-stamp the URL `.../2019/did/v1` because 1) we do want to signal to developers how old the thing they're using is and 2) we may need to publish revisions that are equivalent in semantics, but different in implementation.

Regarding reason `#1` above - developer signalling - We do this for the same reason, the LD Security specs put dates on all cryptographic-y things -- so people know, hmm... I'm using SecureFoo2011 ... and it's 2019... I should probably be using something more modern.

Regarding reason `#2` above - publishing revisions - We may do `.../2019/did/v1` and five years later, may find a subtle but nasty bug, or want to upgrade to JSON-LD 1.2 and want to publish `.../2024/did/v1` ... and at that time, we'll probably have to publish this as well `.../2024/did/v2`. This versioning scheme allows future WGs to potentially make significant breaking changes between v1 and v2 and not affect deployed/legacy implementations. This decoupling is very powerful and ensures that the work isn't saddled with more legacy concerns than necessary when moving forward.

For all of these reasons, we should publish the context at:

https://www.w3.org/2019/did/v1

and the vocabulary at:

https://www.w3.org/2019/did/

**Comment by iherman** [comment]
> We need to get consensus on the last two in order to make progress. I heard concern from @iherman and @philarcher on the latter two items.

To be very clear, as far as I am concerned, I am fairly neutral on both these issues. I just have comments:

- Other communities (that face similar issues with developers averse to RDF) have decided to use `/ns` URI-s (e.g., Web Annotation) and that issue was never raised; the advantage of `/ns` is that it is short and easy to remember, plus:
- I repeat @philarcher's argument on the call: the W3C setup (including the agreements with MIT) is such that the content of `https://www.w3.org/TR/*` and `https://www.w3.org/ns/*` will remain alive in perpetuity, even if W3C disappears at some point. No such commitment exists, afaik, for date space, i.e., for `https://www.w3.org/2019/*`. Whether this is a consideration we want to take on board is not for me to tell. (@philarcher is that correct?)
- On versioning: I am afraid this is a debate that have been raging ever since I remember... But there is a tendency for such vocabularies to stay cast in concrete in practice, even if the content evolves. The foaf namespace URI is still `xmlns.com/foaf/0.1`, and the community has never changed it that after all (although that was the plan, initially). The same for the RDF namespace (ugly as it may be). I realize JSON-LD is a little bit different because the caching issue becomes more explicit, but if there is a commitment that the context file does not change frequently (say, once every 2-3 years) then this should be manageable...

But these are just my comments, not serious concerns. I go with whatever the WG feels comfortable with; at this point, I think it is more important to _decide_ so that we can move on.

**Comment by iherman** [comment]
One more note on versioning, this time in favor of having it explicit in the URL. I do not know whether it may be required, by some DID methods, to store an immutable DID document. If that _is_ the case, then the fact of referring to a stable and also immutable context file does make sense (even if, strictly speaking, even if the context file changes the DID document does not, but the information deduced from it may change, and that may not be acceptable).

**Comment by msporny** [comment]
>  I do not know whether it may be required, by some DID methods, to store an immutable DID document.

Most DID Methods have this requirement, because there are use cases where you have to be able to go back in time for a DID Document and interpret its contents *as it existed in the past*, which means that if the JSON-LD Context were to ever change (semantically), that historical document could get misinterpreted.

**Comment by TallTed** [comment]
Last I knew, the date URIs (`w3.org/yyyy/group/etc`) were based on the inception year of the group publishing the doc, not the year of publication of that doc. This may be relevant to keep in mind in this discussion (or it may have changed without my noticing, in which case, please disregard this comment).

**Comment by iherman** [comment]
> Last I knew, the date URIs (w3.org/yyyy/group/etc) were based on the inception year of the group publishing the doc, not the year of publication of that doc.

Although this is not some sort of a cast-in-concrete rule, it is indeed the way it is usually done.

**Comment by iherman** [comment]
Following @msporny's lead in #76, I try to bring a closure to this issue.

I think we seem to have, by now, a consensus on:

* Host the context on W3C.
* Redirect, during FPWD/WD to github pages for w3c/did-spec... then lock it down (serve it directly from W3C) once we hit PR/REC. (It is up to the editor's whether they prefer to keep the context file in a separate repo or as part of the spec's repo.)
* The context URI includes a version identification.

This changes the consensus level of https://github.com/w3c/did-spec/issues/5#issuecomment-542269659 insofar as the version in the context seems to be not only preferred, but even necessary for some (like @dmitrizagidulin, per https://github.com/w3c/did-spec/issues/5#issuecomment-542271456). I have also removed the CR phase as a possibility to "freeze" the context as the time of freezing this; implementations may come up with issues during CR, so I we should keep the redirect until the publication of the PR. It is only at PR phase that the content of the spec become cast in concrete, and I believe the context file should follow that.

We don't have consensus on:

* What the URL will be, whether or it will be hosted at w3.org/ns/ or a date-space

I will, therefore, add two comments to the issue, use thumbs up/down (and only those) to cast your vote. Only explicit votes will count. I.e., speak now or forever hold your peace...

I would say give this a week before we declare a winner.

**Comment by iherman** [comment]
The context file's URL should be: https://www.w3.org/ns/did/v1

**Comment by iherman** [comment]
The context file's URL should be: https://www.w3.org/2019/did/v1

**Comment by iherman** [comment]
@burnburn @brentzundel has this issue been closed? At the moment there are practically no votes on either way. Would be good to have a decision on this for the FPWD.

(To be clear: it is, process-wise, o.k. to change the context URI in the spec later, but it would not be a good idea to do so. Ie, not finalizing this issue does not mean stopping the FPWD publication.)

Related to https://github.com/w3c/did-core/pull/91

**Comment by brentzundel** [comment]
what about https://www.w3.org/ns/did/2019/v1, that way we have the preservation guarantees of ns AND the date information?

**Comment by iherman** [comment]
@brentzundel that is doable

**Comment by burnburn** [comment]
I like it @brentzundel .  Let's see if any objections before 5 Nov call.  If no serious flaws with it (other than not liking the character string "ns"), we will go with this.

**Comment by msporny** [comment]
> Let's see if any objections before 5 Nov call. If no serious flaws with it (other than not liking the character string "ns"), we will go with this.

-1 for these reasons:

* it is yet another unnecessary variation
* We don't need the preservation guarantees because we include the context and hash in the spec itself.
* it doesn't solve the kneejerk reaction to "ns"

Strongly opposed.

**Comment by yancyribbens** [comment]
I agree with @msporny on this.  It's valuable to add the date to version the URI, however, that's duplicating what `v1` is performing and so it seems unnecessary.

In addition, I think the context should be included in a way such that if the nameserver is unavailable or the JSON-LD processor is offline the context can still be included.  see the following PR for packing the context in a way that's available offline: https://github.com/w3c/vc-data-model/issues/549.  Maybe this should be a new issue instead of piling on to this discussion?

finally, i'm not sure why it matters where the context lives, as long as we all agree on what the hashed value of the resolved context is.  In other words, we can recommend a location `https://www.w3.org/2019/did/v1` but if the same context is resolvable elsewhere with the same hash value, shouldn't that be valid as well?

**Comment by peacekeeper** [comment]
> > I do not know whether it may be required, by some DID methods, to store an immutable DID document.
> 
> Most DID Methods have this requirement, because there are use cases where you have to be able to go back in time for a DID Document and interpret its contents _as it existed in the past_, ...

This is an interesting thought and maybe a bit off-topic for this thread. This should be supported, but I also think that it is trickier than it sounds. Keep in mind that there are also many DID methods that do NOT store an immutable DID document but rather construct one on the fly during resolution. And a DID method may not necessarily be able to know when a DID was created, so it has to make an arbitrary choice which version of the JSON-LD context to use (e.g. the latest).

**Comment by iherman** [comment]
This issue was discussed in [a meeting](https://www.w3.org/2019/did-wg/Meetings/Minutes/2019-11-05-did#section2).

- `RESOLVED:  The DID WG will use <a href="https://www.w3.org/ns/did/v1">https://www.w3.org/ns/did/v1</a> as the URL for the DID spec context documents.`
<details><summary><i class="differentiate">View the transcript</i></summary>
Brent Zundel: <a href="https://github.com/w3c/did-core/issues/5#issuecomment-548783580">https://github.com/w3c/did-core/issues/5#issuecomment-548783580</a><br/>
Brent Zundel: I’m going to give this a 5 minute time box.<br/>
… We’re debating whether we’ll use the ns namespace, there are some contractual guarantees about longevity there vs. using the date field. We don’t seem to have a really clear direction either way. The date field with the ns without the version seems to be slightly more popular.<br/>
… We’d like to fix this sooner rather than later.<br/>
Dave Longley: is the version without ns more popular? I’d like to speak against excluding the version. We found in VC that using the version was useful for comparing the URL against a version that was hashed in the spec, with the expectation that the version would be incremented later<br/>
… if we’re not going ot use a version number, I’m not sure what that would end up looking like. We’ll end up having another bikeshedding discussion for whatever the new thing is going to be<br/>
Ted Thibodeau Jr: straw poll could be useful…<br/>
Daniel Burnett: straw poll has been happening in GitHub<br/>
Ted Thibodeau Jr: there are only a few people participating in github, could we do a straw poll in the call?<br/>
Brent Zundel: Option1: <a href="https://www.w3.org/2019/did/v1">https://www.w3.org/2019/did/v1</a><br/>
Brent Zundel: Option 2: <a href="https://www.w3.org/ns/did/v1">https://www.w3.org/ns/did/v1</a><br/>
Brent Zundel: Option 3: <a href="https://www.w3.org/ns/did/2019/v1">https://www.w3.org/ns/did/2019/v1</a><br/>
Michael Lodder: Option 2<br/>
Yancy Ribbens: +1 option 2<br/>
Amy Guy: 2<br/>
Phil Archer: 2<br/>
Dave Longley: option 2<br/>
Ganesh Annan: 2<br/>
Alexander Hripak: 2<br/>
Joe Andrieu: 2<br/>
David Trang: 2<br/>
Orie Steele: 2<br/>
Michael Jones: 2<br/>
Ted Thibodeau Jr: +0 option1, +1 option2, +0 option3<br/>
Dudley Collinson: Option 2<br/>
Oliver Terbu: 2<br/>
Ivan Herman: 2<br/>
Daniel Buchner: 2<br/>
Markus Sabadello: Examples: VC uses Option 1. ActivityPub uses Option 2.<br/>
Daniel Buchner: Looks like #2 is #1<br/>
Proposed resolution: The DID WG will use <a href="https://www.w3.org/ns/did/v1">https://www.w3.org/ns/did/v1</a> as the URL for the DID spec context documents. (Brent Zundel)<br/>
Ivan Herman: +1<br/>
Brent Zundel: hearing no objections…<br/>
Daniel Burnett: +1<br/>
Amy Guy: +1<br/>
Dave Longley: +1<br/>
Orie Steele: +1<br/>
Joe Andrieu: +1<br/>
Ganesh Annan: +1<br/>
Yancy Ribbens: +1<br/>
Alexander Hripak: +1<br/>
Phil Archer: +1<br/>
David Trang: +1<br/>
Dudley Collinson: +1<br/>
Ted Thibodeau Jr: +1<br/>
Resolution #1: The DID WG will use <a href="https://www.w3.org/ns/did/v1">https://www.w3.org/ns/did/v1</a> as the URL for the DID spec context documents.<br/>
Ivan Herman: To try to make things proper, if one of the editors of the DID core document to make an update that we use this URL everywhere that I can at the last minute I can update the document that will be published on Thursday. If it’s done today then I can handle things tomorrow.<br/>
Amy Guy: I can PR for that if everyone else is busy<br/>
Brent Zundel: Thanks Ivan.<br/>
Markus Sabadello: if <code>@rhiaro</code> does a PR for this, `i can review and merge<br/>
</details>

**Comment by iherman** [comment]
https://www.w3.org/ns/did/v1 is now live, it redirects to the repo's content.

@rhiaro can you check whether it is o.k. for your tests?

@burnburn @brentzundel I propose to close this issue.

**Comment by burnburn** [comment]
Yes.  Once @rhiaro confirms, we can close.

**Comment by msporny** [comment]
Objection to close and objection (as editor) on the resolution. I don't see how the WG was educated on the topic before the straw poll was called, specifically, I can't see where the points against the straw poll item was discussed in the minutes.

**Comment by rhiaro** [comment]
From a content-type perspective it looks like it is served correctly

**Comment by burnburn** [comment]
> Objection to close and objection (as editor) on the resolution. I don't see how the WG was educated on the topic before the straw poll was called, specifically, I can't see where the points against the straw poll item was discussed in the minutes.

@msporny there are no editorial grounds for objection as the resolution was properly taken, since opportunity was made available for anyone on the call to argue for any position they preferred.  Neither Dave Longley nor Dmitri argued in the direction you prefer, so we had no reason to believe you would object at this point.

However, any individual not on the call can object, as you have done, that their position was not sufficiently represented during the discussion.  So as of now, this change is on hold and must be reverted or marked in the document as still under discussion.  If neither of those steps occurs before evening Ivan's time (@iherman ), he has been instructed to halt publication of the FPWD.  Not publishing would be unfortunate.

Personally, I believe that those on the call who care about this issue were aware of your opinions, understood them, and took them into account during the decision.  However, I can't read minds (yet), so we have to resolve this as stated above.

**Comment by dmitrizagidulin** [comment]
@burnburn Minor correction - I was not on that call, otherwise I would have brought up Manu's point.

**Comment by burnburn** [comment]
> @burnburn Minor correction - I was not on that call, otherwise I would have brought up Manu's point.

Apologies.  It doesn't matter at this point, though.  Manu has raised an objection and we will address it or the editors must revert.

**Comment by msporny** [comment]
> If neither of those steps occurs before evening Ivan's time (@iherman ), he has been instructed to halt publication of the FPWD. Not publishing would be unfortunate.

You don't need to hold the FPWD hostage over this... that's extremely bad form.

Go ahead and publish, I'm rescinding my objection because of the unnecessary coupling the chairs and staff have made between this decision and the FPWD.

This decision is going to inevitably kick off a request to do a normative change to the VC spec if the VCWG is re-chartered and has a number of downsides that are problematic (and were not discussed during the call).

**Comment by msporny** [comment]
> Neither Dave Longley nor Dmitri argued in the direction you prefer

These two are different people, with different opinions on matters. We often don't agree with one another (but eventually get there over time). No one should assume that any of us speak for the other.

I'm standing down, publish the FPWD, we'll resolve this later.

**Comment by burnburn** [comment]
> You don't need to hold the FPWD hostage over this... that's extremely bad form.
> 
> Go ahead and publish, I'm rescinding my objection because of the unnecessary coupling the chairs and staff have made between this decision and the FPWD.

Whoa, whoa, whoa.  You objected to a change that went into a document that will be published in a permanently fixed form (FPWD).  Holding off on publishing was done FOR YOU, MANU.  ONLY FOR YOU.
Believe me, no one else is objecting to the current content of the document.  If you are happy for FPWD to go forward regardless of the timing of when this change is backed out, GREAT!

The only coupling here was to avoid YOU complaining that the group went ahead and published an FPWD with a change you didn't agree to.

If you are NOT objecting to that, let's publish the FPWD as planned. If the change is backed out before then, wonderful.  If not, okay.  Either way your objection is in no way being dropped or ignored.

Please feel free to reach out to me or Brent directly if you would like to further discuss your belief that the chairs are in any way trying to force you to do anything.

**Comment by iherman** [comment]
@burnburn great. However, I would propose _not_ to close this issue, in contrast to what we said above.

**Comment by peacekeeper** [comment]
We could add a NOTE or ISSUE box to https://w3c.github.io/did-core/#contexts, stating that this is still under discussion?

**Comment by OR13** [comment]
related... 

https://github.com/w3c/did-core-registry/issues/3
https://github.com/w3c/did-core/issues/187

**Comment by OR13** [comment]
IMO, this is critical blocker, and cannot be closed.

**Comment by msporny** [comment]
My concern here is that the group doesn't have a strategy for versioning and caching and contexts (and registry entries). I don't think we can close this issue w/o understanding what that strategy is... whatever it is. 

So, next step is to publish a concrete strategy for versioning and documenting that thinking in the DID Core implementation guide.

**Comment by dmitrizagidulin** [comment]
@msporny 

> So, next step is to publish a concrete strategy for versioning and documenting that thinking in the DID Core implementation guide.

I agree, and I would propose the text in https://github.com/w3c-ccg/did-spec/pull/272 as the starting document for that versioning strategy.

**Comment by OR13** [comment]
https://github.com/w3c/did-core-registries/issues/17

^ related

**Comment by msporny** [comment]
We've made progress on this:

* DID Core context lives in DID Core Registries.
* W3C ns points to DID Core registries.

We still need to decide:

* Where 3rd party contexts can live.
* Versioning strategy for core context.
* Versioning strategy for extensions context?
* Do we need a suggested versioning strategy for 3rd party extensions?

**Comment by lrosenthol** [comment]
>Where 3rd party contexts can live.

Why can't they live anywhere they want?  It's my context, I'll host it in my environment...

>Do we need a suggested versioning strategy for 3rd party extensions?

Same answer.  It's my extension, let me versioning it any way I want...

**Comment by OR13** [comment]
@lrosenthol yes, i agree. I have added an example of such to the did core registries:

https://w3c.github.io/did-core-registries/#gpgverificationkey2020

**Comment by rhiaro** [comment]
I think this issue can be closed, given development of the DID Spec Registries, which is where the contexts now live.

**Comment by brentzundel** [comment]
Marked `pending close` for more than a week, closing.

**Comment by marek5050** [comment]
Hello all,  curious where is "https://www.w3.org/2018/credentials/v1" now? It resolves to a Github 404 page, so our builds and tests are failing. Should we use another URL? Might be good to update all the documentation and give a warning prior to removing a page like that.

**Comment by iherman** [comment]
This was a temporary glitch with github. It should be all right now.
